File: LeetCode/MergeTwoSortedLinkedList/MergeTwoSortedLists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if(l1 == null) return l2;
        if(l2 == null) return l1;
        ListNode l3 = null;
        ListNode root = null;
        while(l1 != null && l2!=null){
            if(l1.val < l2.val){
                if(l3 == null){
                    l3 = new ListNode(l1.val);
                    root = l3;
                }
                else{
                    ListNode next = new ListNode(l1.val);
                    l3.next = next;
                    l3 = next;
                }
                l1 = l1.next;
                
            }else{
                if(l3 == null){
                    l3 = new ListNode(l2.val);
                    root = l3;
                }
                else{
                    ListNode next = new ListNode(l2.val);
                    l3.next = next;
                    l3 = next;
                }
                l2 = l2.next;
            }
        }
        while(l1 != null){
            ListNode next = new ListNode(l1.val);
            l3.next = next;
            l3 = next;
            l1 = l1.next;
        }
        while(l2 != null){
            ListNode next = new ListNode(l2.val);
            l3.next = next;
            l3 = next;
            l2 = l2.next;
        }
        return root;
    }
    
}

File: LeetCode/MergeTwoSortedLists.py
'''
Merge Two Sorted Lists
https://leetcode.com/explore/learn/card/recursion-i/253/conclusion/2382/

Merge two sorted linked lists and return it as a new list. The new list 
should be made by splicing together the nodes of the first two lists.

Example:

Input: 1->2->4, 1->3->4
Output: 1->1->2->3->4->4

'''
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
    
        if l1 is None:
            return l2
        elif l2 is None:
            return l1
        else:            
            head = ListNode(0)
            l = head
            while l1 is not None and l2 is not None:
                if l1.val < l2.val:                                            
                    l.next = ListNode(l1.val)
                    l1 = l1.next
                    l = l.next
                else:
                    l.next = ListNode(l2.val)
                    l2 = l2.next
                    l = l.next
            if l1 is not None:
                l.next = l1
            else:
                l.next = l2
            return head.next